lib/git_ops/github.ex

```diff
@@ -0,0 +1,90 @@
+defmodule GitOps.GitHub do
+  @moduledoc """
+  GitHub API integration for looking up user information.
+  """
+
+  # alias GitOps.Config
+
+  @cache_key_prefix "git_ops:github_user:"
+
+  @doc """
+  Find a GitHub user by their email address.
+  Returns {:ok, user} if found, where user contains :username, :id, and :url.
+  Returns {:error, reason} if not found or if there's an error.
+
+  Results are cached in persistent_term to avoid repeated API calls.
+  """
+  def find_user_by_email(email) do
+    cache_key = @cache_key_prefix <> email
+
+    case :persistent_term.get(cache_key, :not_found) do
```

Comment:
I think rather than using something like persistent_term, I'd want to go through once at the end with a list of all the GitHub handles, and uniq them and fetch. That would make this logic generally simpler (`:persistent_term` AFAIK should only really be used for effectively "constants", not really as a cache).

Reply:
I agree that persistent_term is a hack. 

for `uniq` emails approach - would have to change more. Will try implementing that part